src/components/Synthetics/PositionEditor/hooks/usePositionEditorData.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  usePositionsConstants,
+  useTokensData,
+  useUserReferralInfo,
+} from "context/SyntheticsStateContext/hooks/globalsHooks";
+import { getLeverage, getLiquidationPrice } from "domain/synthetics/positions";
+import { convertToTokenAmount, convertToUsd } from "domain/synthetics/tokens";
+import { parseValue } from "lib/numbers";
+import { getByKey } from "lib/objects";
+import { useMemo } from "react";
+
+import { useSettings } from "context/SettingsContext/SettingsContextProvider";
+import { usePositionEditorPosition } from "context/SyntheticsStateContext/hooks/positionEditorHooks";
+import { bigMath } from "lib/bigmath";
+import { Operation } from "../types";
+
+import { Options, usePositionEditorFees } from "./usePositionEditorFees";
+
+export function usePositionEditorData({ selectedCollateralAddress, collateralInputValue, operation }: Options) {
+  const { isPnlInLeverage } = useSettings();
+  const tokensData = useTokensData();
+
+  const { minCollateralUsd } = usePositionsConstants();
+  const userReferralInfo = useUserReferralInfo();
+
+  const position = usePositionEditorPosition();
+
+  const isDeposit = operation === Operation.Deposit;
+
+  const collateralToken = getByKey(tokensData, selectedCollateralAddress);
+
+  const collateralPrice = collateralToken?.prices.minPrice;
+
+  const collateralDeltaAmount = parseValue(collateralInputValue || "0", collateralToken?.decimals || 0);
+  const collateralDeltaUsd = convertToUsd(collateralDeltaAmount, collateralToken?.decimals, collateralPrice);
+
+  const { fees } = usePositionEditorFees({
+    selectedCollateralAddress,
+    collateralInputValue,
+    operation,
+  });
+
+  return useMemo(() => {
```

Comment:
could this whole hook be selector instead? it seems it depends only on usePositionEditorFees which can also be a selector if I'm not missing smth

Reply:
Yes, could. I just moved it from PositionEditor's code without any refactoring

Reply:
Made attempt, but there's an issue with `useSettings`. Let's keep this refactoring out of this PR, as it will require some extra-changes not just convert `use*` to `select*`. 

What I did here is just put code from component into separate file.